src/RoboSaverVirtualModule.sol

```diff
@@ -268,15 +295,28 @@ contract RoboSaverVirtualModule {
         /// @dev Last argument `1` stands for `OperationType.DelegateCall`
         delayModule.execTransactionFromModule(MULTICALL3, 0, multicallPayload, 1);
 
-        emit AdjustPoolTxDataQueued(MULTICALL3, abi.encode(calls_));
+        uint256 cachedQueueNonce = delayModule.queueNonce();
+        txQueueData = TxQueueData(cachedQueueNonce, MULTICALL3, multicallPayload);
+
+        emit AdjustPoolTxDataQueued(MULTICALL3, abi.encode(calls_), cachedQueueNonce);
         emit PoolDepositQueued(_card, _surplus, block.timestamp);
 
         return calls_;
     }
 
-    /// @notice Check if there is a transaction queued up in the delay module
+    /// @dev Execute the next transaction in the queue using the storage variable `txQueueData`
+    function _executeNextTx() internal {
+        address cachedTarget = txQueueData.target;
+        delayModule.executeNextTx(cachedTarget, 0, txQueueData.payload, cachedTarget == MULTICALL3 ? 1 : 0);
```

Comment:
maybe make the `1` and `0` more explicit here; so using `Operation.Call` or `Operation.DelegateCall`?

Reply:
could even add a `operation = ...` beforehand to increase readability

Reply:
added the suggestion in 04b456c080290dcce80738efecac2e02835c5d97

initially i was hesitating due to the extra castings needs around those enums. but i think the value add for readability is better

Reply:
why does it need casting to uint8?

Reply:
we are inheriting this [interface](https://github.com/gnosispay/account-kit/blob/main/contracts/interfaces/IDelayModifier.sol#L60)

Reply:
pushed our own interface meantime ff2514133cf1314f286328a27a7bfa8852dba1d1